src/number_theory/padics/padic_norm.lean

```diff
@@ -407,15 +442,13 @@ end padic_val_nat
 
 section padic_val_nat
 
```

Comment:
I would suggest to have an "int" version for all the "padic_val_nat" lemmas. I think number theorists usually prefer working with integers over working with naturals...

(In fact, if I were doing this, I would define padic_val_int first and define padic_val_nat and padic_val_rat in terms of that.)

I have some API lemmas (for my Hilbert symbol code) that look like the following and may fit in here somewhere:
padic_val_int_dvd_iff (p : ℕ) [fact p.prime] (n : ℕ) (a : ℤ) : ↑p^n ∣ a ↔ a = 0 ∨ n ≤ padic_val_int p a
padic_val_int_dvd (p : ℕ) [fact p.prime] (a : ℤ) : ↑p^(padic_val_int p a) ∣ a
padic_val_int_eq_one (p : ℕ) [fact p.prime] : padic_val_int p p = 1
padic_val_int_add_of_mul (p : ℕ) [fact p.prime] {a b : ℤ} (ha : a ≠ 0) (hb : b ≠ 0) :
  padic_val_int p (a*b) = padic_val_int p a + padic_val_int p b
padic_val_int_succ_mul_p (p : ℕ) [fact p.prime] (a : ℤ) (ha : a ≠ 0) : padic_val_int p (a * p) = (padic_val_int p a) + 1

(The names may need some improvement.)

Reply:
Ok, I have implemented all of these, they are around line 585.

Reply:
Thanks!